### PR TITLE
[cwl] Five new files, multimedia.cwl updated, and more

### PR DESCRIPTION
**Pull request by muzimuzhi**
Main changes:
 - Added five new files
    - `incgraph.cwl`
    - `pgffor.cwl` (just for finer dependency of tikz/pgf bundle, see below)
    - `plaintex.cwl` (for storing commands which are available in plain TeX format only)
    - `thmtools.cwl`
    - `xmpmulti.cwl` (one of `beamer`'s subpackages)
 - Reorganized dependency of cwl of packages in tikz/pgf bundle.
 - Updated `multimedia.cwl`, another subpackage of `beamer`.

PS: some of the lines in `xmpmulti.cwl` added and `multimedia.cwl` updated are duplicate with some of the lines in `beamer.cwl`, and this will be fixed when I complete the update for `beamer.cwl` (maybe in next pr).

### CODE REVIEW

File: completion/plaintex.cwl [hunk 1]
```diff
@@ -0,0 +1,4 @@
+\ttfam#*
```
Comment (sunderme) [inline completion/plaintex.cwl:1]: why is this moved to a separate file ?

File: completion/pgffor.cwl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# pgffor package
+# muzimuzhi, 8 Aug 2020, pgffor in tikz v3.1.5b
```
Comment (sunderme) [inline completion/pgffor.cwl:2]: why separate file ?
Is there any scenario where pgffor is separated from pgfcore ?

File: completion/plaintex.cwl (hunk 1 above)
Comment (muzimuzhi) [reply completion/plaintex.cwl:1]: These lines are added to `tex.cwl` in pr #783 opened by myself.

1. All four `\xxfam` commands are macros, not primitive, and are never defined in LaTeX format but only in plainTeX format. 
1. Moreover, I find having them in autocompletion makes the autocompletion less convenient. For example when I type `\ttf` and hit TAB, what I expect is to have `\ttfamily` autocompleted, but with `\ttfam` there I must type more characters to distinguish `\ttfam` from `\ttfamily. (I always set autocompletion "all".)
1. There might be more plainTeX macros currently recorded in `tex.cwl`, but these four are the most annoying ones hence I removed them. 

In most cases, texstudio is used to edit a tex file written in LaTeX format (starting with `\documentclass`), so this change is transparent to most users. PlainTeX users can check `plaintex.cwl` in configuration to add support for plainTeX, hence this change would not remove any utilities.

So in short, remove from `tex.cwl` for LaTeX users, and move to separate `plaintex.cwl` for plainTeX users (and a bit of backward compatibility).

File: completion/pgffor.cwl (hunk 2 above)
Comment (muzimuzhi) [reply completion/pgffor.cwl:2]: Sometimes, yes. As the `pgfmanual` says in sec. 88,

> This package (`pgffor`) can be used independently of PGF. 

Personally I sometimes load `pgffor` without PGF/tikz, mostly when preparing small examples as an answer to online LaTeX questions. In the future I might add more lines to `pgffor.cwl` if possible. Will deeper dependency of cwl files have obvious impact on the user experience?

**Comment by muzimuzhi** [comment]
PS: the CI fails when building on macOS ([link](https://travis-ci.org/github/texstudio-org/texstudio/jobs/715945782#L151))

```
==> cmake . -DBUILD_GTK_TESTS=OFF -DENABLE_CMS=lcms2 -DENABLE_GLIB=ON -DENABLE_QT5=ON -DENABLE_UNSTABLE_API_ABI_HEADERS=ON -DWITH_GObjectIntrospection=ON
==> make install
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

File: completion/pgffor.cwl (hunk 2 above)
Comment (sunderme) [reply completion/pgffor.cwl:2]: Okay, this is fine

File: completion/plaintex.cwl (hunk 1 above)
Comment (sunderme) [reply completion/plaintex.cwl:1]: I am a bit confused why it should be moved from a cwl which is already named tex.cwl.

Comment (sunderme) [reply completion/plaintex.cwl:1]: Any particular reason not use "typical" completion ?

Comment (muzimuzhi) [reply completion/plaintex.cwl:1]: (Sorry it takes me too long to compose and adjust words and sentences to express my intension.)

> I am a bit confused why it should be moved from a cwl which is already named tex.cwl.

Not sure if I understand your confusion, but I guess this is similar to #751.

Concerning the 3 auto-loaded cwl files (`tex.cwl`, `latex-document.cwl` and `latex-dev.cwl`), what they should provide (and provide only) can be split into two parts:
 - the primitives provided by a tex engine, and
 - the commands (including environments) provided by a tex format.

The most common combination is pdftex engine + latex format, and this is what the most lines in those three cwl files do.
 - To add support for engines other than pdftex, separate files [`xetex.cwl`](https://github.com/texstudio-org/texstudio/blob/master/completion/xetex.cwl) and [`luatex.cwl`](https://github.com/texstudio-org/texstudio/blob/master/completion/luatex.cwl) were added years ago. And these two cwl files are not auto-loaded.
 - To add support for formats other than latex,  my decision in pr #783 was to add lines for plainTeX format into `tex.cwl`. But I found this inconvenient, hence in this pr I started to move plainTeX-only lines to separate file `plaintex.cwl`.

This change makes handling for different formats the same for the handling for different engines.

BTW, in practice it is very very rare that one `.tex` file is deliberately made compilable in different formats (compilable both by executable `pdftex` and `pdflatex`), but it is not such rare that one `.tex` file is compilable in different engines. Therefore recording commands for different formats in different files dose, in most cases, no harm.

> Any particular reason not use "typical" completion ?

I need to type in macros marked `#*` in cwl (used for completion only in with the "all" tab) too often. Perhaps I am too far away from typical latex users.

Comment (sunderme) [reply completion/plaintex.cwl:1]: Are you aware that txs counts completion usage and that rare commands appear in typical if they are completed a few times ?

I see that you yourself added these commands ... Are you sure that not even more can be moved ? Default should contain stuff which most users need ...

Comment (muzimuzhi) [reply completion/plaintex.cwl:1]: > Are you aware that txs counts completion usage and that rare commands appear in typical if they are completed a few times ?

Thanks. I will have a try.

> I see that you yourself added these commands ... Are you sure that not even more can be moved ?

There are definitely more can be moved, though currently this pr just moves the most annoying ones. I've found a recent doc ([`tex-nutshell.pdf`](http://mirrors.ctan.org/info/tex-nutshell/tex-nutshell.pdf)) which has a summary of plainTeX macros in sec. 14.

If we both agree that plainTeX-only macros can be moved from `tex.cwl` to `plaintex.cwl`, I can do the rest moving things in one or two days. This way related changes are kept in single pr.

> Default should contain stuff which most users need ...

Most users don't use plainTeX.

Comment (sunderme) [reply completion/plaintex.cwl:1]: Agreed

Comment (muzimuzhi) [reply completion/plaintex.cwl:1]: > [...] I can do the rest moving things in one or two days. This way related changes are kept in single pr.

The moving work has been done in #1203, so I'll mark this conversation as resolved.
